java/com/google/flatbuffers/FlatBufferBuilder.java

```diff
@@ -368,12 +375,30 @@ public int endVector() {
     * @return The offset in the buffer where the encoded string starts.
     */
     public int createString(String s) {
-        byte[] utf8 = s.getBytes(utf8charset);
-        addByte((byte)0);
-        startVector(1, utf8.length, 1);
-        bb.position(space -= utf8.length);
-        bb.put(utf8, 0, utf8.length);
-        return endVector();
+        return createString(getStringBytes(s));
+    }
+
+    private ByteBuffer getStringBytes(String str) {
```

Comment:
Well technically, it's already inlined, that's one thing the JVM excels at.  It's just i think it makes it less clear what the input to ``createString`` is.  Let me known